app/models/pawn.rb

```diff
@@ -1,3 +1,44 @@
 class Pawn < Piece
 
+	def valid_move?(new_x, new_y)
+		[move_number_allowed?(new_y), no_vertical_obstacle?(new_y), not_a_backwards_move?(new_y), diagonal_move_allowed?(new_x, new_y)].all?
+	end
+
+	def move_number_allowed?(new_y)
+		move_number(new_y) == 1 || (move_number(new_y) == 2 && first_move?(new_y))
+	end
+
+	def move_number(new_y)
+		(new_y - y_position).abs
+	end
```

Comment:
This method was added at the suggestion of my mentor. If there is no objection, I will leave it.